util/br/pkg/restore/restore.go

```diff
@@ -0,0 +1,373 @@
+		if err != nil {
+			return err
+		}
+		count--
+	}
+}
+
+func (r *Restore) cleanupOriginal() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := r.backend.RestoreStoragePreCommand(r.config.StorageDataDir + "/nebula")
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	for _, addr := range r.config.MetaAddrs {
+		cmd := r.backend.RestoreMetaPreCommand(r.config.MetaDataDir + "/nebula")
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) stopCluster() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := "cd " + r.config.StorageRootDir + " && scripts/nebula.service stop storaged"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	for _, addr := range r.config.MetaAddrs {
+		cmd := "cd " + r.config.MetaRootDir + " && scripts/nebula.service stop metad"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.MetaUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) startMetaService() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.MetaAddrs {
+		cmd := "cd " + r.config.MetaRootDir + " && scripts/nebula.service start metad &>/dev/null &"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.MetaUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) startStorageService() error {
+	g, _ := errgroup.WithContext(context.Background())
+	for _, addr := range r.config.StorageAddrs {
+		cmd := "cd " + r.config.StorageRootDir + " && scripts/nebula.service start storaged &>/dev/null &"
+		ipAddr := strings.Split(addr, ":")[0]
+		g.Go(func() error { return ssh.ExecCommandBySSH(ipAddr, r.config.StorageUser, cmd, r.log) })
+	}
+
+	err := g.Wait()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restore) RestoreCluster() error {
+
+	err := r.downloadMetaFile()
+	if err != nil {
+		r.log.Error("download meta file failed", zap.Error(err))
+		return err
+	}
+
+	m, err := r.restoreMetaFile()
+
+	if err != nil {
+		r.log.Error("restore meta file failed", zap.Error(err))
+		return err
+	}
+
+	err = r.checkPhysicalTopology(m.BackupInfo)
+	if err != nil {
+		r.log.Error("check physical failed", zap.Error(err))
+		return err
+	}
+
+	err = r.stopCluster()
+	if err != nil {
+		r.log.Error("stop cluster failed", zap.Error(err))
+		return err
+	}
+
+	err = r.cleanupOriginal()
```

Comment:
Better to move to the final step

Reply:
We have to leave it here for now, because we are downloading the meta and storage backups directly to their data directories, and after this step, we will download the content directly from the remote.